src/rules/alwaysUppercaseID.ts

```diff
@@ -0,0 +1,46 @@
+import { ESLintUtils } from "@typescript-eslint/utils";
+
+export const alwaysUppercaseID = ESLintUtils.RuleCreator.withoutDocs({
+    create(context) {
+        return {
+            Identifier(node) {
+                const name = node.name;
+
+                if (
+                    (name.endsWith("Id") || name.endsWith("Ids")) &&
+                    !(name.endsWith("ID") || name.endsWith("IDs"))
+                ) {
+                    let found = "";
+
+                    if (name.endsWith("id")) {
+                        found = "id";
+                    } else if (name.endsWith("ids")) {
+                        found = "ids";
+                    } else if (name.endsWith("Id")) {
+                        found = "Id";
+                    } else if (name.endsWith("Ids")) {
+                        found = "Ids";
+                    }
+
+                    context.report({
+                        node,
+                        messageId: "useUppercaseID",
+                        data: { found },
+                    });
+                }
+            },
+        };
+    },
```

Comment:
Handle ID in middle of variablename. Something like:

```suggestion
function splitCamelCase(identifier: string): string[] {
    return identifier.split(/(?=[A-Z])/);
}


export const alwaysUppercaseID = ESLintUtils.RuleCreator.withoutDocs({
    create(context) {
        return {
            Identifier(node) {
                const name = node.name;
                const parts = splitCamelCase(name);

                for (const part of parts) {
                    if (part === "Id" || part === "Ids") {
                        context.report({
                            node,
                            messageId: "useUppercaseID",
                            data: { found: part },
                        });
                        break; // only report once per identifier
                    }
                }
            },
        };
    },
```

Reply:
Yeah but this way we loose the check for when ID is not followed by a camelcase, like userId123. Not very common I'd say, but still a case. I do not think we do this anywhere. But still a case.